api/v1alpha1/aws_node_types.go

```diff
@@ -105,3 +113,48 @@ type AMILookup struct {
 	// +kubebuilder:validation:MaxLength=32
 	BaseOS string `json:"baseOS,omitempty"`
 }
+
+// +kubebuilder:validation:XValidation:rule="has(self.root) || size(self.nonroot) > 0",message="either root or nonroot must be specified"
```

Comment:
Thinking about this it seems redundant. It's not that if one of these isn't specified then something will break so neither is strictly required. This just adds processing for no actual functional benefit.
```suggestion
```

Reply:
👍 Im good with reverting the commit too